File: src/taeeun/week1/양궁대회.js
function solution(n, info) {
    let answer=new Array(11).fill(0);

    const calcScore=(ryan, apeach)=>{
        let ryanScore=0;
        let apeachScore=0;
        for (let i=0; i<10; i++) {
            if (ryan[i]==0 && apeach[i]==0) continue;
            if (ryan[i]>apeach[i]) ryanScore+=(10-i);
            else apeachScore+=(10-i);
        }
        if (apeachScore>=ryanScore) return -1;
        else return ryanScore-apeachScore;
    }
    
    const isLower=(current, ryan)=>{
        for (let i=10; i>=0; i--) {
            if (current[i]>ryan[i]) return false;
            if (current[i]<ryan[i]) return true;
        }
        return false;
    }
    
    let maxDiff=-1;
    
    const solve=(apeach, ryan, current, arrows)=>{
        //남은 화살이 없거나 끝난 경우 점수 계산
        if (arrows==0 || current>10) {
            let diff=calcScore(ryan, apeach);
            if (diff==-1) return;
            ryan[10]+=arrows; // 낮은 점수를 더 많이 맞혀야하니까 0점에 남은 화살 더함
            
            //더 큰 점수 차이인지
            //낮은 점수를 더 많이 맞혔는지
            if (diff == maxDiff&&isLower(answer, ryan)) {
                answer=ryan;
            }
            else if (diff > maxDiff) {
                answer=ryan;
                maxDiff=diff;
            }
            return;
        }
        
        //점수를 가져가는 경우
        if (apeach[current] < arrows) {
            const temp=[...ryan]
            temp[current]=apeach[current]+1;
            solve(apeach, temp, current+1, arrows-(apeach[current]+1));
        }
        //점수를 가져가지 않는 경우
        solve(apeach, ryan, current+1, arrows);
    }
    
    solve(info, answer, 0, n);

    if (maxDiff==-1) return [-1];
    else return answer;
}

File: src/chanhyun/1week/양궁대회.py
from itertools import combinations_with_replacement

def compare(apeach, ryan):
    r = 0
    a = 0
    for i in range(11):
        if ryan[i] == 0 and apeach[i] == 0:
            continue
        if ryan[i] > apeach[i]:
            r += 10-i
        else:
            a += 10-i
    if r>a:
        return r-a
    else:
        return -1

def solution(n, info):
    answer = []
    max_result = 0
    for case in combinations_with_replacement([i for i in range(11)], n):
        count = [0 for _ in range(11)]
        for c in case:
            count[10-c] += 1
        result = compare(info, count)
        if result > max_result:
            max_result = result
            answer = count
    if answer == []:
        return [-1]
    else:
        return answer

########################

from itertools import combinations_with_replacement

def compare(apeach, ryan):
    apeach_score = 0
    ryan_score = 0
    for i in range(11):
        if apeach[i] == 0 and ryan[i] == 0:
            continue
        if apeach[i] >= ryan[i]:
            apeach_score += 10-i
        else:
            ryan_score += 10-i
    if apeach_score < ryan_score:
        return [apeach_score, ryan_score]
    else:
        return []

def solution(n, info):
    answer = []
    score_diff = -1
    points = [i for i in range(11)]
    cases = list(combinations_with_replacement(points, n))
    ryans = []
    for c in cases:
        p = [0 for _ in range(11)]
        for x in c:
            p[10-x] += 1
        ryans.append(p)
    for r in ryans:
        result = compare(info, r)
        if result != []:
            if result[1]-result[0] > score_diff:
                score_diff = result[1]-result[0]
                answer = r
    if score_diff == -1:
        return [-1]
    else:
        return answer